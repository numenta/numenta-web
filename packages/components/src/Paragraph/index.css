/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

/* dropcap - sync with wrappers/md/md.css */
._dropcap:first-of-type::first-letter {
  background-color: #fff;
  color: #1aa0db;
  float: left;
  font-weight: 500;
  line-height: 0.85;
  margin: 0 0 -1rem 0;
  /* -s (<= mobile) */
  font-size: 6.8rem;
  padding: 0.5rem 0.3rem 0.3rem 0;
}
@media screen and (min-width: 48em) {
  ._dropcap:first-of-type::first-letter {
    /* -ns (> mobile) */
    font-size: 5.4rem;
    padding: 0.4rem 0.3rem 0.3rem 0;
  }
}


.paragraph {
  composes: lh-copy from 'tachyons';
  composes: measure-wide from 'tachyons';
  composes: measure-m from 'tachyons';
  composes: measure-wide-l from 'tachyons';
  composes: mt0 from 'tachyons';
}

.lead {
  composes: _dropcap;
}
