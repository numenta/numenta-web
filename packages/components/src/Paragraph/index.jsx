// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


const Paragraph = ({children, lead}) => {
  const classes = [styles.paragraph]

  if (lead === true) classes.push(styles.lead)

  return (
    <p className={classNames(...classes)}>
      {children}
    </p>
  )
}

Paragraph.propTypes = {
  children: React.PropTypes.node,
  lead: React.PropTypes.bool,
}

Paragraph.defaultProps = {
  lead: false,
}

export default Paragraph
