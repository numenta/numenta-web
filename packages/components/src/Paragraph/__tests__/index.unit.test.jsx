import React from 'react'
import renderer from 'react-test-renderer'

import Paragraph from '../../Paragraph'


describe('Paragraph React component', () => {

  it('Renders Lead correctly', () => {
    const component = renderer.create(
      <Paragraph lead={true}>
        Headline
      </Paragraph>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Default correctly', () => {
    const component = renderer.create(
      <Paragraph lead={false}>
        Contents
      </Paragraph>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
