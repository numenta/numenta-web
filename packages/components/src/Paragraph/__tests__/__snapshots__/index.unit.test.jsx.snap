exports[`Paragraph React component Renders Default correctly 1`] = `
<p
  className="paragraph">
  Contents
</p>
`;

exports[`Paragraph React component Renders Lead correctly 1`] = `
<p
  className="paragraph lead">
  Headline
</p>
`;
